### DISCUSSION TITLE
formulae or casks on M2 Silicon

### DISCUSSION THREAD

**Discussion by rajivkapoor1** [Homebrew/discussions#4635] [Writing Formulae/Casks]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

Hi, I am just trying to clarify whether the brew that is installed for M1/M2 Silicon, always will download the version of the Macbook formulae/casks that are NATIVELY built for M1/M2 Silicon OR whether they just install the Intel versions of the formulae/casks. Even if it is the Intel versions of the formulae/casks, my understanding is that they will still work on M1/M2 Silicon, via the use of Rosetta 2 emulating Intel and running them on M1/M2.

In other words of phrasing my question in more specific terms, I can use the example of say installing VirtualBox cask, using Homebrew, to explain what I mean when I ask this. So, here goes....

I understand that if one downloads VirtualBox on the Macbook from the webiste (and NOT using brew), there is a specific version for Intel macbooks, and another version for M1/M2 Silicon. So, one can choose which one is suitable and download away.

If though, one is using brew, and brew , as we know, when running on the Intel Macbooks, it installs packages into /usr/local/Cellar (I think?), and brew on M1/M2 Silicon Macbooks installs packages into /opt/homebrew/Cellar. So, lets assume for this question, we are using brew on the M1/M2 Silicon, and all packages will be installed in /opt/homebrew/Cellar.

The question though, is , if we use brew on Silicon to install VirtualBox, will it ALWAYS install the NATIVE version of VirtualBox? Or will it just install the Intel version (which will probably work OK although not perfectly via Rosetta 2).

**Comment by SMillerDev** [comment]
> The question though, is , if we use brew on Silicon to install VirtualBox, will it ALWAYS install the NATIVE version of VirtualBox? Or will it just install the Intel version (which will probably work OK although not perfectly via Rosetta 2).

Seeing how VirtualBox does not have stable native binaries for ARM, this is not an example I can give a good general answer on.

---
Generally the answer is this: a native brew install will always try to install native binaries.
An install in `/opt/homebrew` on ARM will install native formulae and native casks where they are available. If they are not available they can instead install the x86 based binaries.
